### PR TITLE
Support DPI change in sizers

### PR DESCRIPTION
**Pull request by MaartenBent**
Return the size of `DoGetDefaultBorderInPx` as double, so no precision is lost when multiplying it for `DoubleBorder` and `TripleBorder`.

Closes [#18551](https://trac.wxwidgets.org/ticket/18551).

### CODE REVIEW

File: include/wx/sizer.h [hunk 1]
```diff
@@ -223,7 +228,7 @@ class WXDLLIMPEXP_CORE wxSizerFlags
 
 private:
 #ifdef wxNEEDS_BORDER_IN_PX
-    static int DoGetDefaultBorderInPx();
+    static double DoGetDefaultBorderInPx();
```
Comment (PBfordev) [inline include/wx/sizer.h]: Just curious, you use `double` here and in `wxPrivate::DpiDependentValue<double>` while `float` seems to be used elsewhere for scaling with DPI?

File: src/msw/window.cpp [hunk 2]
```diff
@@ -4884,6 +4884,38 @@ wxWindowMSW::MSWUpdateOnDPIChange(const wxSize& oldDPI, const wxSize& newDPI)
...
+    // update sizers
+    if ( GetSizer() )
+    {
+        wxSizerItemList::compatibility_iterator current =
+            GetSizer()->GetChildren().GetFirst();
+        while ( current )
+        {
+            wxSizerItem* sizerItem = current->GetData();
+
+            int borderOld = sizerItem->GetBorder();
```
Comment (PBfordev) [inline src/msw/window.cpp]: Is `borderOld` really needed?

**Review by PBfordev — COMMENTED** [review]
Thanks again.

File: src/msw/window.cpp (hunk 2 above)
Comment (MaartenBent) [reply src/msw/window.cpp]: No, its a remnant from my debugging output.

File: include/wx/sizer.h (hunk 1 above)
Comment (MaartenBent) [reply include/wx/sizer.h]: `GetContentScaleFactor()` returns double so I kept using that. I can use float if this is preferred.

Comment (PBfordev) [reply include/wx/sizer.h]: I see, sorry for the noise.

Comment (MaartenBent) [reply include/wx/sizer.h]: It did make me look at the code again, and I noticed `C6011: Dereferencing NULL pointer 'win'.` IntelliSense warning. Every remark can be useful.

File: include/wx/sizer.h [hunk 3]
```diff
@@ -106,6 +106,11 @@ class WXDLLIMPEXP_CORE wxSizerFlags
 
     // default border size used by Border() below
     static int GetDefaultBorder()
+    {
+        return wxRound(GetDefaultBorderDouble());
+    }
+
+    static double GetDefaultBorderDouble()
```
Comment (vadz) [inline include/wx/sizer.h]: It's a nitpick but I wonder if we can use something better than "Double" in the new function name. Unfortunately I don't have any great ideas myself, something like `GetDefaultBorderFractionalPx()` is probably too long, but maybe still worth it?

Another idea would be to add `static int GetScaledDefaultBorder(int multiple)` that would avoid using `double` in the interface completely and use them only internally.

File: src/msw/window.cpp [hunk 4]
```diff
@@ -4884,6 +4884,38 @@ wxWindowMSW::MSWUpdateOnDPIChange(const wxSize& oldDPI, const wxSize& newDPI)
     // update font if necessary
     MSWUpdateFontOnDPIChange(newDPI);
 
+    // update sizers
+    if ( GetSizer() )
+    {
+        wxSizerItemList::compatibility_iterator current =
+            GetSizer()->GetChildren().GetFirst();
+        while ( current )
```
Comment (vadz) [inline src/msw/window.cpp]: I prefer writing these loops as
```
        for ( wxWindowList::compatibility_iterator node = GetChildren().GetFirst();
              node;
              node = node->GetNext() )
```
but it's pretty minor, of course. A `for` loop would be simpler to replace with ranged for in the future...

**Review by vadz — COMMENTED** [review]
Thanks!

As usual, I'm fine with applying this as is, except that the new function needs to be documented.

But please let me know if you see any merit in either of my proposals.

File: include/wx/sizer.h (hunk 3 above)
Comment (MaartenBent) [reply include/wx/sizer.h]: I will rename it. It can also cause confusion with `DoubleBorder()` where double has a different meaning.

**Comment by vadz** [comment]
Thanks again!

**Pull request merged**